pkg/controller/node/node_controller_test.go

```diff
@@ -429,129 +439,72 @@ func TestGetCandidateMachines(t *testing.T) {
 		},
 		expected: nil,
 	}, {
-		//progress all available nodes
+		// node-2 is going to change config, so we can only progress one more
 		progress: 2,
 		nodes: []*corev1.Node{
 			newNodeWithReady("node-0", "v1", "v1", corev1.ConditionTrue),
-			newNodeWithReady("node-1", "v1", "v1", corev1.ConditionFalse),
+			newNodeWithReady("node-1", "v1", "v1", corev1.ConditionTrue),
```

Comment:
do we have coverage on the flip above though? or it can't never happen with this new changes?

Reply:
Mmmm...I think I'd answer that question by saying it's a better test if we indeed keep one unready node, and make it look like this:

```
diff --git a/pkg/controller/node/node_controller_test.go b/pkg/controller/node/node_controller_test.go
index 74e951df..0a772379 100644
--- a/pkg/controller/node/node_controller_test.go
+++ b/pkg/controller/node/node_controller_test.go
@@ -440,10 +440,10 @@ func TestGetCandidateMachines(t *testing.T) {
 		expected: nil,
 	}, {
 		// node-2 is going to change config, so we can only progress one more
-		progress: 2,
+		progress: 3,
 		nodes: []*corev1.Node{
 			newNodeWithReady("node-0", "v1", "v1", corev1.ConditionTrue),
-			newNodeWithReady("node-1", "v1", "v1", corev1.ConditionTrue),
+			newNodeWithReady("node-1", "v1", "v1", corev1.ConditionFalse),
 			newNodeWithReady("node-2", "v0", "v1", corev1.ConditionTrue),
 			newNodeWithReady("node-3", "v0", "v0", corev1.ConditionTrue),
 			newNodeWithReady("node-4", "v0", "v0", corev1.ConditionTrue),
```

will push after this current CI run if you agree.